pom.xml

```diff
@@ -50,8 +50,8 @@
 
     <properties>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-        <maven.compiler.source>1.8</maven.compiler.source>
-        <maven.compiler.target>1.8</maven.compiler.target>
+        <maven.compiler.source>17</maven.compiler.source>
+        <maven.compiler.target>17</maven.compiler.target>
```

Comment:
I wonder if you could leverage `maven.compiler.release`.

https://stackoverflow.com/questions/59049980/maven-compiler-release-as-an-replacement-for-source-and-target